README.md

```diff
@@ -93,6 +93,47 @@ Requires `ActiveDirectoryLDAPConfig(domain: String, url: String, searchFilter: S
    - For JDK11: Run `keytool -import -file ldapcert.pem -alias ldaps -keystore <path_to_jdk>/lib/security/cacerts -storepass <password>` (default password is *changeit*).
 5. Enter `yes` when prompted.
 
+## Key Provider
+The Application allows for the user to allow the application to generate a key in memory.
+This is useful for single deployments and testing, however, may present issues when trying to deploy multiple login-services for redundancy.
+To get around this, the application allows for you to generate your keys in AWS Secrets manager and the application will periodically fetch them.
+
+In order to setup for in-memory key generation, your config should look like so:
+```
+loginsvc:
+  rest:
+    jwt:
+      generate-in-memory:
+        exp-time: 4
+        alg-name: "RS256"
+```
+The important information to provide is exp-time which indicates how long a token is valid for
+and alg-name which indicates which algorithm is used to encode your keys.
+
+To setup for AWS Secrets Manager, your config should look like so:
+```
+loginsvc:
+  rest:
+    jwt:
+      fetch-from-aws:
+        secret-name: "secret"
+        region: "region"
+        private-aws-key: "privateKey"
+        public-aws-key: "publicKey"
+        exp-aws-key: "expiry"
+        alg-aws-key: "alg"
+        refresh-time: 30
```

Comment:
The `exp-time: 4` and `refresh-time: 30` do not seem clear on the time unit used. Perhaps just include a comment _in hours_, _in days_. I am trying to solve this in https://github.com/AbsaOSS/login-service/pull/78 for fields like these to be a `FiniteDuration` type.

If you merge sooner, I can apply the solution for your duration values or vice-versa

Reply:
Added FiniteDuration Types for Config